File: codes/Question5_1/C/Question5_1.c
#include <stdio.h>
#include <stdlib.h>
#include "Staff.h"
#include "CLinkedList.h"

Staff* WhoIsOnDutyNext(List* plist, char* name, int days_later)
{
	Staff* pstaff = (Staff*)malloc(sizeof(Staff));
	int i;

	if (LFirst(plist, &pstaff))
	{
		if (!strcmp(pstaff->name, name))
		{
			for (i = 0; i < days_later; i++)
			{
				LNext(plist, &pstaff);
			}
			return pstaff;
		}

		while (1)
		{
			LNext(plist, &pstaff);
			if (!strcmp(pstaff->name, name))
			{
				for (i = 0; i < days_later; i++)
				{
					LNext(plist, &pstaff);
				}
				return pstaff;
			}

		}
	}
}

int main(void)
{
	List list;
	Staff* pstaff;

	ListInit(&list);

	// 4명의 사원 정보 저장
	pstaff = (Staff*)malloc(sizeof(Staff));
	SetStaffInfo(pstaff, 2201036, "정바름");
	LInsert(&list, pstaff);

	pstaff = (Staff*)malloc(sizeof(Staff));
	SetStaffInfo(pstaff, 2003014, "나유연");
	LInsert(&list, pstaff);

	pstaff = (Staff*)malloc(sizeof(Staff));
	SetStaffInfo(pstaff, 1903048, "구아름");
	LInsert(&list, pstaff);

	pstaff = (Staff*)malloc(sizeof(Staff));
	SetStaffInfo(pstaff, 2003075, "신재현");
	LInsert(&list, pstaff);

	// 누구의 당직인지 확인
	pstaff = WhoIsOnDutyNext(&list, "정바름", 6);
	ShowStaffInfo(pstaff);

	return 0;
}

File: codes/Question5_1/Python/Question5_1.py
import StaffModule as sm
import CircularLinkedListModule as clm

def WhoIsOnDutyNext(clist, name, days_later):
    data = clist.LFirst()
    if data[0] is True:
        if data[1].name is name:
            for i in range(days_later):
                data = clist.LNext()
            return data[1]
    
    while True:
        data = clist.LNext()
        if data[1].name is name:
            for i in range(days_later):
                data = clist.LNext()
            return data[1]

clist = clm.CircularLinkedList()

# 4명의 사원 정보 저장
clist.LInsert(sm.Staff(2201036, "정바름"))
clist.LInsert(sm.Staff(2003014, "나유연"))
clist.LInsert(sm.Staff(1903048, "구아름"))
clist.LInsert(sm.Staff(2003075, "신재현"))

# 누구 당직인지 확인
WhoIsOnDutyNext(clist, "정바름", 6).ShowStaffInfo()
